### PR TITLE
Fixes cmake build & c++(17) compile errors for windows/vc++ 2019

### PR DESCRIPTION
**Pull request by korejan**
Hi, I re-did my fixes for cmake build issues & C++(17) compiler errors for windows & visual C++ 2019  for Hyperpose.

This version is a little better and this time I didn't include my changes to fix the not-existing unistd.h file on this platform for std tracer. I did try to do in a better way that @lgarithm suggested [here](https://github.com/tensorlayer/hyperpose/issues/251#issuecomment-631378840), parameterized the git tag like for std tensor, defaulting to tag v0.2.1 however this version of std tracer does not generate a static library (tracer.cpp is not part of the cmake files anymore) which causes linker errors . I couldn't figure out away in cmake to override this in the EXTERNALPROJECT_ADD command in 3rdparty.cmake as i don't know enough about cmake.

So for now in my local repo I just manually copy a unistd.h file I made into 3rdparty\src\stdtracer-repo\include

### CODE REVIEW

**Comment by lgarithm** [comment]
the stdtracer library is now a pure header only library, and I assume user should define the extern variables in their libraries, or executables, by using the macro `DEFINE_TRACE_CONTEXTS` in one of their `*.cpp` files.

I did a temporary fix example in #267, could you test if that works for your case? Thanks.

File: examples/utils.cpp [hunk 1]
```diff
@@ -1,6 +1,15 @@
 #include "utils.hpp"
 
+#if !((defined(__cplusplus) && (__cplusplus >= 201703L)) || (defined(_MSVC_LANG) && (_MSVC_LANG >= 201703L)))
```
Comment (lgarithm) [inline examples/utils.cpp:3]: How about introduce a file `include/hyperpose/filesystem` and put all the `#if ...` in side it.
So all other files in this project can just `#include <hyperpose/filesystem>`.

e.g. 

define a customized  `filesystem` header
https://github.com/stdml/stdtensor/blob/master/tests/include/ttl/filesystem

and use it in another file without the `#if...` macros.
https://github.com/stdml/stdtensor/blob/ef1a5249fa04880e20bcd18df69c2d73ede41db8/tests/test_loc.cpp#L3

**Comment by korejan** [comment]
> 
> 
> the stdtracer library is now a pure header only library, and I assume user should define the extern variables in their libraries, or executables, by using the macro `DEFINE_TRACE_CONTEXTS` in one of their `*.cpp` files.
> 
> I did a temporary fix example in #267, could you test if that works for your case? Thanks.

I wanted to try this now but I'm blocked becuase of [this issue](https://github.com/tensorlayer/hyperpose/issues/268)

File: examples/utils.cpp (hunk 1 above)
Comment (korejan) [reply examples/utils.cpp:3]: Could do but I wonder if any of this is really necessary, if all the required C++ compilers (and versions of them) for hyperpose has C++17 standard conformance and C++17 is the current standard may aswell just not use the std::experimental version of filesystem.

**Comment by lgarithm** [comment]
It seems my branch doesn't contain the bad file. So this might work `git clone https://github.com/tensorlayer/hyperpose -b lg-trace` before we fix the issue on master.

**Comment by korejan** [comment]
@lgarithm I've merged those changes into my master fork branch, builds & tested with debug & release. Should I make another pull request?

**Comment by lgarithm** [comment]
@korejan I think you can still use this PR. Would you like me to merge it?

**Comment by korejan** [comment]
> 
> 
> @korejan I think you can still use this PR. Would you like me to merge it?

@lgarithm Yeah that would good thanks.

**Pull request merged**